### PR TITLE
Update to fuel-core=0.7, fuel-asm=0.5, fuel-tx=0.10, fuel-types=0.5, …

### LINKED ISSUE #1620: Update fuel-core version to 0.7.0
[fuel-core v0.7.0](https://github.com/FuelLabs/fuel-core/releases/tag/v0.7.0) is released. To use that we will need to update the following:

- fuel-types 0.5
- fuel-vm 0.9
- fuel-crypto 0.5
- fuel-tx 0.10
- fuel-asm 0.5

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #1620

### CODE REVIEW

**Review by adlerjohn — COMMENTED** [review]
CI failing

**Comment by kayagokalp** [comment]
I think we also need a new release of [fuels-rs](https://github.com/FuelLabs/fuels-rs/). I am converting this to draft until we have a new release of [fuels-rs](https://github.com/FuelLabs/fuels-rs/) that uses new versions of fuel-* (fuel-vm, fuel-types, etc.)

File: test/src/sdk-harness/Cargo.toml [hunk 1]
```diff
@@ -7,10 +7,10 @@ version = "0.0.0"
 
 [dependencies]
 fuel-core = { version = "0.6", default-features = false }
```
Comment (mohammadfawaz) [inline test/src/sdk-harness/Cargo.toml:9]: Should this be bumped to `0.7` as well?

Comment (kayagokalp) [reply test/src/sdk-harness/Cargo.toml:9]: Oh, I guess I missed this one, thanks :)

**Comment by mohammadfawaz** [comment]
Note that I've already had to bump `fuel-asm` for a different PR. You'll probably notice that as soon as you merge with master :)

**Comment by adlerjohn** [comment]
Superseded by #1683

**Pull request closed without merging**